src/main/groovy/com/blackducksoftware/integration/hub/detect/bomtool/npm/NpmNodeModulesDependencyFinder.groovy

```diff
@@ -0,0 +1,154 @@
+				return createNodeFromPackageNode(pNode, nodeModulesFile)
+			} else {
+				logger.info("No package.json found, parsing node_modules folder")
+				return createNodeFromNodeModules(nodeModulesFile)
+			}
+		} else if(pNode) {
+			/*
+			 * We have only the package.json and no node_modules folder which would require REST calls to complete.
+			 * This would be a good place to check for shrinkwrap/package-lock files
+			 */
+			logger.error("No node_modules folder located in target path. Can't run program")
+		}
+	}
+
+	/*
+	 * The best case scenario for parsing npm without npm installed (uses package.json and node_modules).
+	 * Note: Looks to be faster than the Cli finder
+	 */
+	public DependencyNode createNodeFromPackageNode(PackageNode pNode, File nodeModulesFile) {
+		DependencyNode node = createDependencyNode(pNode)
+
+		//Had to create a HashSet object before adding content because of issues with the keySet method for maps
+		def allDependencies = new HashSet<String>()
+		allDependencies.addAll(pNode?.dependencies?.keySet())
+		Set<String> optionalDependencies = pNode?.optionalDependencies?.keySet()
+
+		if(optionalDependencies) {
+			allDependencies.addAll(optionalDependencies)
+		}
+
+		for(String filename : allDependencies) {
+			if(modules.contains(filename)) {
+				continue
+			} else {
+				modules.add(filename)
+				def packageFile = fileFinder.findFile(nodeModulesFile.getAbsolutePath() + '/' + filename + '/', NpmConstants.PACKAGE_JSON)
+				if(packageFile) {
+					node.children.add(createNodeFromPackageNode(deserializeJsonToPackageNode(packageFile), nodeModulesFile))
+				}
+			}
+		}
+
+		node
+	}
+
+	/*
+	 * A slower check through the node_modules directory that doesn't require a starting package.json for the tree.
+	 */
+	public DependencyNode createNodeFromNodeModules(File nodeModulesFile) {
+		def pNode = new PackageNode();
+		pNode.name = 'BrianMandel'
+		pNode.version = new Date().format('MM.dd.yy')
+
+		final DependencyNode node = createDependencyNode(pNode)
+
+		nodeModulesFile.eachDir() { dir ->
+			def packageFile = fileFinder.findFile(dir.getAbsolutePath() + '/', NpmConstants.PACKAGE_JSON)
+			if(packageFile) {
+				DependencyNode tempNode = createNodeFromPackageNode(deserializeJsonToPackageNode(packageFile), nodeModulesFile)
+
+				node.children.add(tempNode)
+			}
+		}
+
+		node
+	}
+
+	private DependencyNode createDependencyNode(PackageNode pNode) {
+		def name = pNode.name
+		def version = pNode.version
+		def externalId = new NameVersionExternalId(Forge.NPM, name, version)
+
+		def node = new DependencyNode(name, version, externalId)
+	}
+
+	public PackageNode deserializeJsonToPackageNode(File depOut) {
+		def fileReader
+		def jsonReader
+		try {
+			fileReader = new FileReader(depOut)
+			jsonReader = new JsonReader(fileReader)
+		} catch (Exception e) {
+			println(depOut)
+			e.printStackTrace()
+		}
+		def packageNode = (PackageNode) gson.fromJson(jsonReader, PackageNode.class)
```

Comment:
depOut.text will get you the text contents of the file in groovy.  Its a faster way to read the file but can not be set to use a specific encoding but it doesn't seem we are worried about that